plan/resolve_indices.go

```diff
@@ -31,18 +33,26 @@ func (p *PhysicalHashJoin) ResolveIndices() {
 	p.physicalSchemaProducer.ResolveIndices()
 	lSchema := p.children[0].Schema()
 	rSchema := p.children[1].Schema()
-	for _, fun := range p.EqualConditions {
+	for i, fun := range p.EqualConditions {
+		fun = fun.Clone().(*expression.ScalarFunction)
```

Comment:
why `clone` here?

Reply:
Because we need to clone it before we call `ResolveIndices`.